interoperator/api/resource/v1alpha1/sfcluster_types.go

```diff
@@ -17,23 +17,74 @@ limitations under the License.
 package v1alpha1
 
 import (
+	"k8s.io/apimachinery/pkg/api/resource"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
+// Resources specifies the resource types which are used by schedulers
+// to schedule instances to clusters
+type Resources struct {
+	CPU    resource.Quantity `yaml:"cpu,omitempty" json:"cpu,omitempty"`
+	Memory resource.Quantity `yaml:"memory,omitempty" json:"memory,omitempty"`
+}
+
+// Equal return true if Resources is equal to y. Otherwise returns false
+func (r *Resources) Equal(y *Resources) bool {
+	if r == nil {
+		if y == nil {
+			return true
+		}
+		return false
+	}
+	if y == nil {
+		return false
+	}
+	return r.CPU.Equal(y.CPU) && r.Memory.Equal(y.Memory)
+}
+
+// Less returns true if the Resource is less to y, Otherwise return false
+func (r *Resources) Less(y *Resources) bool {
+	if y == nil {
+		return false
+	}
+	if r == nil {
+		return true
+	}
+
+	cpu := r.CPU.Cmp(y.CPU)
+	memory := r.Memory.Cmp(y.Memory)
+	return cpu == -1 && memory == -1
+}
+
 // SFClusterSpec defines the desired state of SFCluster
 type SFClusterSpec struct {
 	// Name of the secret containing the kubeconfig required to access the
 	// member cluster. The secret needs to exist in the same namespace
 	// as the SFCluster and should have a "kubeconfig" key.
-	SecretRef string `json:"secretRef"`
+	SecretRef string `yaml:"secretRef" json:"secretRef"`
+
+	// Capacity represents the total resources of a cluster.
+	Capacity *Resources `yaml:"capacity,omitempty" json:"capacity,omitempty"`
+
+	// +kubebuilder:validation:Minimum=0
+	// +kubebuilder:validation:Maximum=100
+	// Determines the how filled the cluster becomes, before interoperator filters out the cluster as full.
+	BufferPercentage int `yaml:"bufferPercentage,omitempty" json:"bufferPercentage,omitempty"`
```

Comment:
Wouldn't it be better to call this `limitThreshold`, `limitThresholdPercentage`, `schedulingLimit` or `schedulingLimitPercentage` or something like that? The term buffer is ambiguous about whether it refers to `1 - actual/total` or `actual/total`.

Reply:
Renamed to `schedulingLimitPercentage`